pallets/gear/src/lib.rs

```diff
@@ -1677,6 +1688,59 @@ pub mod pallet {
 
             Ok(())
         }
+
+        /// Pay additional rent for the program.
+        #[pallet::call_index(8)]
+        #[pallet::weight(<T as Config>::WeightInfo::pay_rent())]
+        pub fn pay_rent(
+            origin: OriginFor<T>,
+            program_id: ProgramId,
+            block_count: BlockNumberFor<T>,
+        ) -> DispatchResultWithPostInfo {
+            let who = ensure_signed(origin)?;
+
+            ProgramStorageOf::<T>::update_active_program(
+                program_id,
+                |program| -> Result<(), Error<T>> {
+                    let block_author =
+                        Authorship::<T>::author().ok_or(Error::<T>::BlockAuthorNotFound)?;
```

Comment:
This error should be unreachable, isn't it?

Reply:
It maybe however it is easier to just return an error from the extrinsic

Reply:
`unwrap_or_else unreachable`

Reply:
plus extract function